src/executables/sysrepo-plugind.c

```diff
@@ -323,13 +327,93 @@ load_plugins(struct srpd_plugin_s **plugins, int *plugin_count)
         /* finally store the plugin */
         (*plugins)[*plugin_count].handle = handle;
         (*plugins)[*plugin_count].private_data = NULL;
+        {
+            const char *src = ent->d_name;
+            const size_t src_len = strlen(src);
+            const size_t n = strcmp(src + (src_len - 3), ".so") == 0 ?
+                src_len - 3 : src_len;
+            char *str = strndup(src, n);
+            if (!str) {
+                error_print(0, "strndup() failed.");
+                dlclose(handle);
+                rc = -1;
+                break;
+            }
+            (*plugins)[*plugin_count].plugin_name = str;
+        }
```

Comment:
Please do not use blocks just cause of variable definitions, it is confusing. I always prefer having all the variables at the function beginning but it is fine if you add something later unless a whole block is required like this. Then put it at the beginning.